Tutor times frontend Dashboard for tutors and Visualisation Using Ngx Graphs

EkamBhullar opened the pull request:
# Description

## Tutor Times Feature Implementation

###  Overview
This PR implements the Tutor Times component for tracking and analyzing tutor assessment activities.
<img width="2716" height="1578" alt="image" src="https://github.com/user-attachments/assets/e2f3bfc3-3330-44c9-8acd-4a1aad36703c" />


###  Features
- **Assessment Summary**: Overview of sessions, time spent, and tasks assessed
- **Task Analytics**: Time spent per task with visual charts
- **Session Monitoring**: Real-time tracking of active marking sessions
- **Activity Timeline**: Detailed session activities and assessments

###  UI/UX
- Material UI components for consistency
- Tailwind CSS for responsive design
- Clean, modern interface optimized for tutors

### Technical Details
- Angular component with TypeScript interfaces
- Performance optimizations with trackBy functions
- Memory leak prevention with OnDestroy
- Dummy data for development and testing

###  Future Integration
- Ready for backend API integration when available
- Interface definitions prepared for backend schema
- Easy switch from dummy to real data


## Type of change


- [x] New feature (non-breaking change which adds functionality)


# How Has This Been Tested?

- Login as a Tutor
-  Choose Tutor Times from the dropdown
-  You can view the dashboard and test if it renders properly,
-  View Details of each sessions
## Testing Checklist:

- [x] Tested in latest Chrome
- [x] Tested in latest Safari

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

zara1504 committed:
chore: remove media-service.coffee and modify doubtfire-angularjs.module.ts

EkamBhullar committed:
feat: add visualtisation for  tutor-times frontend

EkamBhullar committed:
feat: add visualtisation for  tutor-times frontend

Pasindufdo98 reviewed (commented):
Hi @EkamBhullar ,
I tested your new frontend feature locally and UI looks clean and the core flows behave as expected. Nice work on the layout and consistency with the existing pattern library.

I did hit one console error:

`NG0303: Can't bind to 'matRowDefTrackBy' since it isn't a known property of 'tr'`

Happy to take another look once this is fixed.

<img width="1920" height="1133" alt="Screenshot (1864)" src="https://github.com/user-attachments/assets/28aaf015-4e43-4a30-9582-6eee9ffef8f0" />

<img width="1920" height="1123" alt="Screenshot (1866)" src="https://github.com/user-attachments/assets/b7be12fc-bbf8-4a5f-844d-dfe2b0492eda" />

chelaz1234 commented:
Hi @EkamBhullar,
Nice addition! The Tutor Times dashboard slots neatly into the header menu, and the UI copy reads clearly. Using ngx-charts is a good fit for quick visual summaries. However, I’m also seeing the same console error that Pasindu mentioned. Could you please fix it and let me know?
<img width="1918" height="962" alt="image" src="https://github.com/user-attachments/assets/7926f32b-31b2-4b08-82f9-55ba35eeaab9" />

EkamBhullar committed:
fix(tutor-times): remove trackBy from *matRowDef to resolve binding error

EkamBhullar commented:
Hey @Pasindufdo98 and @chelaz1234 . Thanks For the review! I have updated the code accordingly. Thanks

returnMarcco committed:
Migrate/group set selector (#981)

* chore: migrate group-set-selector (WIP)

- define and declare new component files

- import new component into app

- downgrade component

* chore: migrate group-set-selector

- convert AngularJS template syntax to Angular syntax
- finish implementing `group-set-selector.component.ts`
- remove redundant Coffeescript file

* chore: migrate group-set-selector

- finish defining Typescript logic
- update parent component to use Angular syntax for child component declaration

* chore: cleanup and documentation

- remove redundant Coffeescript file
- adjust frontend migration todo list

* chore: remove redundant file

- remove redundant Coffeescript file

* fix: attribute and event binding

- fix attribute and event binding syntax
- remove redundant template file

* refactor: two-way binding

- remove redundant directive attributes

- rename EventEmitter var name in child component

* Update README.md

* Update README.md

* refactor: replace Bootstrap components with Angular Material

- replace Bootstrap components with Angular Material components & directives

- refactor to use modern Angular structural directives

returnMarcco committed:
refactor: migrate/date service (#931)

* chore: date-service.coffee to TS

- add `date.service.ts` file
- add `date.service.spec.ts` unit test
- downgrade TS module such that it works within AngularJS

* chore: unlink date-service.coffee

- complete unlinking process of  `date-service.coffee` after migrating to Typescript
- manual linting/comment formatting

* chore: delete redundant coffeescript file

- delete `date-service.coffee` file after migration

* chore:  formatting

- format code with prettier

lachlan-robinson committed:
refactor: migrate progress dashboard (#927)

* chore: 9.x windows

* chore: create new component files

* chore: unlink old component + link new component

* chore: migrate functionality + update to tailwindcss and material UI

* chore: remove old component files

* chore: responsive columns for visualisations

* Update package-lock.json

* Update package.json

* Update package.json

* Update package-lock.json

* chore: update styling to better match original

* chore: add custom breakpoint for responsive change

* chore: tweak breakpoint for better visualisation visibility

* chore: update tutor view functionality + tweak event emitter name

---------

Co-authored-by: Boink <40929320+b0ink@users.noreply.github.com>

lachlan-robinson committed:
fix: burndown chart visualisation (#942)

* fix: update burndown data to render series of different lengths

* chore: cleanup imports

* chore: remove leftover testing comment from previous contributor

b0ink committed:
Merge pull request #929 from zara1504/remove/media-service

Remove Media Service

lachlan-robinson committed:
refactor: migrate task status pie chart (#928)

* chore: migrate pie chart to ngxchart

* chore: remove testing console log

* chore: restore package and lock to 9.x

* chore: remove testing comments

* chore: resolve merge conflicts

* chore: remove testing console log

* chore: resolve futher conflicts

* chore: exlude packge-lock

* chore: exclude package

* chore: delete migrated progress dashboard html

---------

Co-authored-by: Boink <40929320+b0ink@users.noreply.github.com>

BrianDangDev commented:
Hey @EkamBhullar , need 2 approvals before I can approve it

Pasindufdo98 reviewed (approved):
<img width="1918" height="1137" alt="image" src="https://github.com/user-attachments/assets/856dcace-7e45-4d55-9d16-c568352321f5" />
Good work @EkamBhullar . No more browser console errors.

chelaz1234 reviewed (approved):
Hi @EkamBhullar,

Good work overall! Every function works as expected, and no errors are coming up in the console either.

EkamBhullar commented:
Hey @BrianDangDev , I have got 2 approvals. Thanks

BrianDangDev reviewed (approved):
Good work, please remember to include details about the dummy data in the documentation, so others are aware this feature currently uses mock data and will need backend integration before production.

ibi420 reviewed (changes requested):
Hey @EkamBhullar , I can confirm this feature works as expected. You committed both the package-lock.json and package.json files only the package.json needs to be committed. Also, I noticed you’re trying to merge into development; could you switch that to 9.x instead? Aside from these small changes, everything looks good. Nice work on this.

EkamBhullar committed:
chore: remove package-lock.json

EkamBhullar committed:
chore: ignore package-lock.json

EkamBhullar committed:
chore: resolve 9.x merge conflicts; integrate Tutor Times and update modules/states

EkamBhullar committed:
feat(tutor-times): declare TutorTimesComponent in Angular module

EkamBhullar committed:
chore: resolve lockfile merge conflict by removing package-lock.json

EkamBhullar committed:
chore: resolve 9.x merge conflicts

EkamBhullar committed:
chore: merge upstream 9.x; drop lockfile; keep upstream modules
